### PR TITLE
EOS-26653: Added submodule dispatch event

### PR DESCRIPTION
**Pull request by venkuppu-chn**
Signed-off-by: Venkatesh K <venkatesh.k@seagate.com>

# Problem Statement
RE's submodule is missing in parent repository

# Design
Whenever code push in main branch the submodule event is dispatched to add it to parent repository

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [x] Unit and System Tests are added
- [x] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Review by shailesh-vaidya — APPROVED** [review]

**Comment by shailesh-vaidya** [comment]
@gauravchaudhari02 Can you please review this.

File: .github/workflows/dispatch_submodule_update.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+name: Dispatch event for cortx submodule
+on:
+  push:
+    branches: [ main ]
```
Comment (gauravchaudhari02) [inline .github/workflows/dispatch_submodule_update.yml:20]: @venkuppu-chn, as we are raising this PR for `legacy-main` branch, should it be `legacy-main` as well?
CC - @shailesh-vaidya

Comment (venkuppu-chn) [reply .github/workflows/dispatch_submodule_update.yml:20]: @gauravchaudhari02 I'm not getting you. Do we need to raise the same PR against legacy-main not main branch?

Comment (gauravchaudhari02) [reply .github/workflows/dispatch_submodule_update.yml:20]: image
You are raising PR towards `legacy-main` but dispatch event will trigger on `main` branch as per code.

**Comment by venkuppu-chn** [comment]
@gauravchaudhari02 I have created new PR due to conflict and DCO failure.

https://github.com/Seagate/cortx-re/pull/821

**Pull request closed without merging**